Create the initial data model

tijmenb opened the pull request:
### Context

We're about to kick off work on the vendor API and candidate form. We'll need a data model to start off with.

### Changes proposed in this pull request

See commits messages for details.

### Guidance to review

Review per-commit.

### Link to Trello card

https://trello.com/c/NLqwffuf/994-create-the-initial-data-model-in-the-application

duncanjbrown commented on README.md:
```diff
@@ -60,3 +61,11 @@ check` before proceeding to debug using `bin/webpack`.
 by invoking `bin/webpack`. If all is well, there is a chance that
 `public/packs-test` contains stale output. Delete it and re-run the suite.
+
+## Documentation
```
`Documentation` now has equal status with `Webpacker` and `Docker workflow`. Should we demote those to be under a `Development process` heading or sth?

benilovj commented on adr/003-initial-datamodel.md:
```diff
@@ -0,0 +1,32 @@
+- `ApplicationForm`, which belongs to a Candidate, and holds the
+information that is common to all of the applications that the user
+makes (in the "Apply 1" phase, they can apply for up to three courses).
+- `CourseChoice`, which holds the information that is course specific,
```
I have some reservations about this class name.

Based on the name, I erroneously assumed that this model would model the choice of the course (e.g. the course and the training location) – I've even independently used this name [on the prototype branch](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/208).

duncanjbrown replied:
I've put a quick meeting in at 3:30 this afternoon for us to discuss this

duncanjbrown commented on spec/models/candidate_spec.rb:
```diff
@@ -6,4 +6,18 @@
   it { is_expected.to validate_uniqueness_of :email_address }
+
+  describe '#delete' do
+    it 'deletes all dependent records through cascading deletes in the database' do
```
Happy to leave it here if it's the best/only place to document it

tvararu commented on config/database.yml:
```diff
@@ -10,6 +10,7 @@ default: &default
 
 development:
   <<: *default
+  database: 'bat_apply_development'
```
Looks fine to me too; Publish uses these names: https://github.com/DFE-Digital/manage-courses-backend/blob/master/config/database.yml#L10-L24

chubberlisk commented on adr/003-initial-datamodel.md:
```diff
@@ -0,0 +1,32 @@
+# 2. Initial data model
```
Smol change, we're on our third ADR.

```suggestion
# 3. Initial data model
```

tijmenb commented:
Updated after an offline discussion with @tvararu @benilovj and @duncanjbrown. It turns out that "course choice" is an existing concept, meaning the combination of a course and a training location (so, "something you can choose").

We've renamed our model to "Application Choice", as in "the choice that you've made on your application". Added some additional comments to the models, added the ucas attributes to the schema, and listed some words in the README.

chubberlisk commented on adr/0003-initial-datamodel.md:
```diff
@@ -0,0 +1,32 @@
+
+The eventual table structure will look something like this:
+
+![data model](003-initial-datamodel.png)
```
```suggestion
![data model](0003-initial-datamodel.png)
```

Doesn't seem to display I think it's because it's unable to find the image, possibly because of the above.
